runners/spark/src/main/java/org/apache/beam/runners/spark/translation/TransformTranslator.java

```diff
@@ -414,6 +400,38 @@ public void evaluate(ParDo.BoundMulti<InputT, OutputT> transform, EvaluationCont
         }
       }
 
+      private void evaluateSingle(
```

Comment:
When I try to condense the two, java yells at me:

Caused by: java.io.NotSerializableException: DStream checkpointing has been enabled but the DStreams with their functions are not serializable
org.apache.beam.runners.spark.translation.EvaluationContext

I don't know how valuable it is to investigate this. I'm comfortable with commenting that this both has better performance and also file a JIRA about condensing the two if appropriate.

I have factored out the common parts in both this and the streaming translator, and filed https://issues.apache.org/jira/browse/BEAM-1737

@amitsela Any ideas why the serialization failure would occur?